Runtime/CourseController/CourseControllerSetup.cs

```diff
@@ -35,12 +34,32 @@ private void CreateCourseController()
             }
             else
             {
-                Type courseControllerType = ReflectionUtils.GetTypeFromAssemblyQualifiedName(courseControllerQualifiedName);
+                Type courseControllerType = RetrieveCourseControllerType();
+                
                 if (ReflectionUtils.CreateInstanceOfType(courseControllerType) is ICourseController controller)
                 {
-                    courseController = Instantiate(controller.GetCourseControllerPrefab());
+                    Instantiate(controller.GetCourseControllerPrefab());
                 }
             }
         }
+
+        private Type RetrieveCourseControllerType()
+        {
+            if (string.IsNullOrEmpty(courseControllerQualifiedName))
+            {
+                return RetrieveDefaultControllerType();
+            }
+            
+            Type courseControllerType = ReflectionUtils.GetTypeFromAssemblyQualifiedName(courseControllerQualifiedName);
+            return courseControllerType != null ? courseControllerType : RetrieveDefaultControllerType();
+        }
+
+        private Type RetrieveDefaultControllerType()
+        {
+            return ReflectionUtils.GetFinalImplementationsOf<ICourseController>()
+                .Select(c => (ICourseController) ReflectionUtils.CreateInstanceOfType(c)).OrderByDescending(controller => controller.Priority)
+                .First()
```

Comment:
There is chance that there is no implementation of ICourseController existing, so there is still a possible NPE

Reply:
I take it back, we are in ui components ^^

Reply:
😅